sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -55,61 +57,133 @@ private[hive] class HiveFunctionRegistry(
     }
   }
 
-  override def lookupFunction(name: String, children: Seq[Expression]): Expression = {
-    Try(underlying.lookupFunction(name, children)).getOrElse {
-      // We only look it up to see if it exists, but do not include it in the HiveUDF since it is
-      // not always serializable.
-      val functionInfo: FunctionInfo =
-        Option(getFunctionInfo(name.toLowerCase)).getOrElse(
-          throw new AnalysisException(s"undefined function $name"))
+  def loadHivePermanentFunction(name: String): Option[CatalogFunction] = {
+    val databaseName = sessionStage.catalog.getCurrentDatabase
+    val func = FunctionIdentifier(name, Option(databaseName))
+    val catalogFunc =
+      if (sessionStage.catalog.listFunctions(databaseName, name).size != 0) {
```

Comment:
also, I'm not sure if this is the responsibility of `HiveFunctionRegistry`. Let's talk about that in the main thread.